### PR TITLE
Add a gitolite source

### LINKED ISSUE #27: gitolite source
Hi friend :)

I'd like to add a gitolite source and wanted to run the approach by you before sending a PR. A couple questions:

1. For getting the list of repositories, seems like the only way provided via the gitolite API itself is [the `info` command](http://gitolite.com/gitolite/user/#the-info-command). It provides a handy `-json` flag, e.g.:
   ```
   $ ssh git@git.mycompany.com info -json
   {
      "repos" : {
         "a_repo" : {
            "perms" : {
               "W" : 1,
               "R" : 1
            }
         },
         "some_repo-DEPRECATED" : {
            "perms" : {
               # "W" key is omitted entirely
               "R" : 1
            }
         }
      }
      "gitolite_version" : "1.2.3",
      "USER" : "git@somehost",
      "GL_USER" : "ckuehl",
      "git_version" : "1.2.3"
   }
   ```

   Potentially we could provide other ways to get this list (e.g. something like y/gits), but I'm thinking if I wanted to do that, I'd just format that into a JSON file and use the JSON file source.

   So, does over SSH sound okay? And if so, do you prefer subprocessing out to ssh or using paramiko? (I think I prefer the subprocess approach just to avoid adding a new dep?)

2. Actually I guess I only had one question. This looks pretty straightforward to implement :P

### PR DESCRIPTION
**Pull request by chriskuehl**
Fixes #27

### CODE REVIEW

**Comment by asottile** [comment]
Seems good, one thing to think about (that wasn't really considered when only github was a source) is what happens when paths are nested.

gitolite enables repositories as follows:

```
./repo1
./dir/repo2
./dir/repo2/actually_another_repo
```
As `all-repos` is currently implemented, this would clobber `actually_another_repo` *into* `repo2`.

I *think* I solved this in the private yelpy all_repos by appending `.git` onto the repo name -- thoughts?

**Comment by chriskuehl** [comment]
Good point, I like that idea.

File: README.md [hunk 1]
```diff
@@ -96,6 +96,29 @@ Clones all repositories available to a user on github.
...
+#### Optional `source_settings`
+
+- `mirror_path` (default `None`): an optional mirror to clone repositories from.
+  This is a Python format string, and can use the variable `repo_name`.
+
+  This can be anything git understands, such as another remote server (e.g.
+  `gitmirror.mycompany.com:{repo_name}`) or a local path (e.g.
+  `/gitolite/git/{repo_name}.git`).
+
 #### Directory location
```
Comment (asottile) [inline README.md]: woops, this spliced the github documentation -- fixing :D

Comment (asottile) [reply README.md]: https://github.com/asottile/all-repos/commit/b688501dd5e3041d314bcc71dbe1026925e524a8

Comment (chriskuehl) [reply README.md]: derp, thanks!

**Pull request merged**